main.go

```diff
@@ -251,11 +251,17 @@ func computeRoutePrefix(prefix string, externalURL *url.URL) string {
 	}
 
 	if prefix == "/" {
-		prefix = ""
+		return ""
 	}
 
-	if prefix != "" {
-		prefix = "/" + strings.Trim(prefix, "/")
+	// Ensure prefix starts with "/"
+	if !strings.HasPrefix(prefix, "/") {
+		prefix = "/" + prefix
+	}
+
+	// Ensure prefix does not end with "/"
+	if strings.HasSuffix(prefix, "/") {
+		prefix = strings.TrimSuffix(prefix, "/")
```

Comment:
If you look at the [source code of `strings.TrimSuffix`](https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/strings/strings.go;l=1052), the first line in the function is literary `if HasSuffix(s, suffix) {`, so you don't really need L263 here.